prime-router/src/main/kotlin/Receiver.kt

```diff
@@ -91,6 +92,7 @@ open class Receiver(
         routingFilter: ReportStreamFilter = emptyList(),
         processingModeFilter: ReportStreamFilter = emptyList(),
         conditionFilter: ReportStreamFilter = emptyList(),
+        mappedConditionFilter: ReportStreamConditionFilter = emptyList(),
```

Comment:
Good catch 👍 this is intended to be temporary. The `ReportStreamConditionFilter` is a new abstract class that will implement filtering behavior. `ReportStreamFilter` is just type aliased String.

In this PR we are leaving the `conditionFilter`, and an upcoming PR will 'combine' the `mappedConditionFilter` and `conditionFilter` to be the same thing. It was left separate for now to avoid a large configuration change and lower the chance of breaking existing functionality.